src/librustc_metadata/creader.rs

```diff
@@ -550,9 +550,10 @@ impl<'a> CrateLoader<'a> {
         dep_kind: DepKind,
     ) -> CrateNumMap {
         debug!("resolving deps of external crate");
-        if crate_root.is_proc_macro_crate() {
-            return CrateNumMap::new();
-        }
```

Comment:
Ah, yes, this should have been done as a part of https://github.com/rust-lang/rust/pull/69432.
It causes issues though, like you can see from the CI failures.

Could you remove this change from this PR and submit it as a separate PR.

Reply:
Opened https://github.com/rust-lang/rust/pull/69976

Reply:
Can this PR be merged without this change or some tests fail?
If it can, then could you remove this change from this PR.

Reply:
@petrochenkov: Unfortunately, it can't. I added this because `librustc_span` fails to compile without it: https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/23051/logs/71

Reply:
Spans with out-of-range positions could be decoded as dummy in the meantime.

Reply:
The ICE occurs when we try to decode the `CrateNum`, so we would need to manually decode an integer with the same size as `CrateNum` to avoid throwing off metadata decoding. If you think that's worth it, I'll add it in.

Reply:
Yeah, something like that may be reasonable to do, it's a small hack.
Before https://github.com/rust-lang/rust/pull/69432 `CrateNum`s of unloaded crates were mapped into their "parent" crates - https://github.com/rust-lang/rust/pull/69432/files#diff-974185a1d0122c6d970a1e4faaedbe5fL541-L552.
Proc macro dependencies can do the same thing.
```rust
        if crate_root.is_proc_macro_crate() {
            return vec![krate; crate_root.decode_crate_deps(metadata).count() + 1];
        }
```
, basically.

Reply:
I've added in this hack